src/targets/gpu/kernels/migraphx_kernels.cpp

```diff
@@ -0,0 +1,87 @@
+        HMODULE handle{ ::GetModuleHandle(nullptr) };
+        HRSRC rc{ ::FindResource(handle, MAKEINTRESOURCE(id), MAKEINTRESOURCE(TEXTFILE)) };
+        HGLOBAL data{ ::LoadResource(handle, rc) };
+        return { static_cast<char const *>(::LockResource(data)),
+                ::SizeofResource(handle, rc) };
+    }
+}
+
+std::vector<src_file> migraphx_kernels()
+{
+    static src_file _kernels_[] = {
+        { "migraphx/kernels/algorithm.hpp", resource::read(IDR_ALGORITHM_HPP) },
+        { "migraphx/kernels/args.hpp", resource::read(IDR_ARGS_HPP) },
+        { "migraphx/kernels/array.hpp", resource::read(IDR_ARRAY_HPP) },
+        { "migraphx/kernels/concat.hpp", resource::read(IDR_CONCAT_HPP) },
+        { "migraphx/kernels/debug.hpp", resource::read(IDR_DEBUG_HPP) },
+        { "migraphx/kernels/dfor.hpp", resource::read(IDR_DFOR_HPP) },
+        { "migraphx/kernels/dpp.hpp", resource::read(IDR_DPP_HPP) },
+        { "migraphx/kernels/functional.hpp", resource::read(IDR_FUNCTIONAL_HPP) },
+        { "migraphx/kernels/gather.hpp", resource::read(IDR_GATHER_HPP) },
+        { "migraphx/kernels/gathernd.hpp", resource::read(IDR_GATHERND_HPP) },
+        { "migraphx/kernels/generic_constant.hpp", resource::read(IDR_GENERIC_CONSTANT_HPP) },
+        { "migraphx/kernels/hip.hpp", resource::read(IDR_HIP_HPP) },
+        { "migraphx/kernels/index.hpp", resource::read(IDR_INDEX_HPP) },
+        { "migraphx/kernels/integral_constant.hpp", resource::read(IDR_INTEGRAL_CONSTANT_HPP) },
+        { "migraphx/kernels/iota_iterator.hpp", resource::read(IDR_IOTA_ITERATOR_HPP) },
+        { "migraphx/kernels/layernorm.hpp", resource::read(IDR_LAYERNORM_HPP) },
+        { "migraphx/kernels/math.hpp", resource::read(IDR_MATH_HPP) },
+        { "migraphx/kernels/ops.hpp", resource::read(IDR_OPS_HPP) },
+        { "migraphx/kernels/pad.hpp", resource::read(IDR_PAD_HPP) },
+        { "migraphx/kernels/pointwise.hpp", resource::read(IDR_POINTWISE_HPP) },
+        { "migraphx/kernels/preload.hpp", resource::read(IDR_PRELOAD_HPP) },
+        { "migraphx/kernels/print.hpp", resource::read(IDR_PRINT_HPP) },
+        { "migraphx/kernels/ranges.hpp", resource::read(IDR_RANGES_HPP) },
+        { "migraphx/kernels/reduce.hpp", resource::read(IDR_REDUCE_HPP) },
+        { "migraphx/kernels/roialign.hpp", resource::read(IDR_ROIALIGN_HPP) },
+        { "migraphx/kernels/scatternd.hpp", resource::read(IDR_SCATTERND_HPP) },
+        { "migraphx/kernels/shape.hpp", resource::read(IDR_SHAPE_HPP) },
+        { "migraphx/kernels/softmax.hpp", resource::read(IDR_SOFTMAX_HPP) },
+        { "migraphx/kernels/tensor_view.hpp", resource::read(IDR_TENSOR_VIEW_HPP) },
+        { "migraphx/kernels/type_traits.hpp", resource::read(IDR_TYPE_TRAITS_HPP) },
+        { "migraphx/kernels/types.hpp", resource::read(IDR_TYPES_HPP) },
+        { "migraphx/kernels/vec.hpp", resource::read(IDR_VEC_HPP) },
+        { "migraphx/kernels/vectorize.hpp", resource::read(IDR_VECTORIZE_HPP) }
+    };
```

Comment:
These files should not be added to the repo. They should just be generated by cmake in the build directory.

Reply:
It is a temporary solution. We have already added a Jira ticket to work on that. We need something to start working on enabling tests, etc.

Reply:
BTW why this file needs to be generated? On Windows, resource file format supports including external files (binary and text). We do not need to generate anything.

Reply:
> It is a temporary solution. We have already added a Jira ticket to work on that. We need something to start working on enabling tests, etc.

Yea hopefully #1804 can help get it working for windows and then we can work on trying to get it to work with the resource files later on.

> why this file needs to be generated? 

We shouldn't need to update source files when we add new file to embed, that should be handled automatically with the `add_embed_library` function.